### PR TITLE
Patch windows

### PR DESCRIPTION
**Pull request by pudymody**
Hi, i found your plugin on Reddit and i liked its simplicity! Good job.
Sadly in my try to test it, i found out that it wasnt able to execute on windows.

Here is a patch for that.
I dont know if EVERYTHING is working, but basic navigation, creation of folder/files, copy/cut and deletion was working on my tests.

It also checks that the current version of Neovim is at least 0.5, otherwise it doesnt load itself.

Fixes: #208

### CODE REVIEW

**Comment by pudymody** [comment]
I've just found out that git integration doesnt work, so i think its not ready to merge yet.
Im sorry i havent tested it enough.

**Comment by kyazdani42** [comment]
cool thanks :) tell me when you're done i'll review the code still works on linux too !

**Comment by pudymody** [comment]
Now everything should work as expected.

File: lua/nvim-tree/git.lua [hunk 1]
```diff
@@ -53,6 +60,11 @@ local function create_root(cwd)
     return false
   end
 
+	local is_win = vim.api.nvim_call_function("has", {"win32"})
+	if is_win then
+		git_root = git_root:gsub("/", "\\")
+	end
+
```
Comment (kyazdani42) [inline lua/nvim-tree/git.lua:67]: this file is not properly indented

**Review by kyazdani42 — CHANGES_REQUESTED** [review]
I think the path_separator code is called many times, it could be loaded only once and accessed through it's own module, providing functions like `join_path`. Also why package.config instead of `vim.fn.has('win32') and '\\' or '/'` ? could windows have `/` path separators ?

**Comment by pudymody** [comment]
> Also why package.config instead of `vim.fn.has('win32') and '\\' or '/'`

No particular reason, i just prefer lua to tell me what the separator is, instead of defining it myself.

> could windows have / path separators ?

In windows the '/' character is used to separate folders/files, but i liked the consistency of having all the paths with '\\' and not mix between both.

As per the module, is it okay if i put it as a function inside the utils module?

**Comment by kyazdani42** [comment]
allright, yes go for it

**Comment by pudymody** [comment]
Done :)

File: lua/nvim-tree/git.lua [hunk 2]
```diff
@@ -60,10 +60,10 @@ local function create_root(cwd)
     return false
   end
 
-	local is_win = vim.api.nvim_call_function("has", {"win32"})
-	if is_win then
-		git_root = git_root:gsub("/", "\\")
-	end
+  local is_win = vim.api.nvim_call_function("has", {"win32"})
```
Comment (kyazdani42) [inline lua/nvim-tree/git.lua:63]: you could call this function only once, not on very create_root call :)

**Review by kyazdani42 — CHANGES_REQUESTED** [review]
one last small thing and we're good to go. I'm testing this PR on linux in the meanwhile to make sure nothing was broke

**Comment by kyazdani42** [comment]
This break git integration on linux not sure why

**Comment by pudymody** [comment]
You were absolutely right, i was silly and forgot to check that the condition was true. (i thought that it returned an integer).
Now everything should be working again.

**Comment by kyazdani42** [comment]
looks good, thanks for working on that. You can also update the README and the doc file to remove the `not working on windows` information and i'll merge this :)

**Comment by kyazdani42** [comment]
thanks for the contribution !

**Pull request merged**